python/F3DPythonBindings.cxx

```diff
@@ -14,13 +14,38 @@
 
 namespace py = pybind11;
 
+template<typename T, size_t S>
+void pyseq_to_array(const py::sequence& seq, std::array<T, S>& arr)
+{
+  const size_t len = py::len(seq);
+  if (len != arr.size())
+    throw py::value_error("wrong length");
+  for (size_t i = 0; i < len; ++i)
+    arr[i] = py::cast<T>(seq[i]);
+}
+
 PYBIND11_MODULE(f3d, module)
 {
   module.doc() = "f3d library bindings";
 
   // types
-  py::class_<f3d::point3_t>(module, "point3_t").def(py::init<double, double, double>());
-  py::class_<f3d::vector3_t>(module, "vector3_t").def(py::init<double, double, double>());
+  py::class_<f3d::point3_t>(module, "point3_t")
+    .def(py::init<double, double, double>())
+    .def("__len__", [](const f3d::point3_t& p) { return p.size(); })
+    .def(
+      "__iter__", [](const f3d::point3_t& p) { return py::make_iterator(p); },
+      py::keep_alive<0, 1>()) /* keep alive while iterator is used */
+    .def("__getitem__", [](const f3d::point3_t& p, int i) { return p[i]; })
+    .def("__setitem__", [](f3d::point3_t& p, int i, double value) { p[i] = value; });
```

Comment:
I have no idea what it does but I'll suppose you know what you are doing :)

Reply:
I think I do but I'm open to be proven wrong :)

- `__getitem__` and `__setitem__` are for subscripting (`x = point[0]`, `point[1] = y`)
- `__iter__` makes the object iterable, the basic usage would be to do `for value in my_collection: ...` which isn't really useful for point/vectors, but it's also used internally when you do `tuple(point)` so in this case you can convert a `point3_t` to a raw tuple in a pythonic way (without needing a `point.to_tuple()` method for example). The language also lets you explode iterables so here you could do `x,y,z = point` to extract the coords in 3 separate variables
- `__len__` is for the built-in `len(obj)` function, here `len(point)` should always return 3. I'm not sure where (or if at all) it is used internally but I left it in anyway, doesn't seem to hurt

Reply:
oh and `py::keep_alive<0, 1>()` is just your regular deep magic incantation you do because the internet said so, as per tradition. Couldn't tell you how that one works or if we _really_ need it.

Reply:
ok thanks!